### ISSUE TITLE
support Jaccard metric in binary index

### ISSUE THREAD

**Issue by Vector1iu** [facebookresearch/faiss#3134]
Subject: Request for New Feature - support Jaccard metric in binary index

Dear Development Team,

I am a user of faiss with great interest. While using your project, I have found that while faiss support many metric type, but faiss does not support Jaccard metric in binary index, while [knowhere](https://github.com/zilliztech/knowhere/blob/main/thirdparty/faiss/faiss/utils/jaccard-inl.h) support it.

I believe this feature would be very useful for all users of your project. If you deem it valuable, I would be extremely grateful.

Thank you for your time and consideration.

Best regards,

**Comment by mlomeli1** [comment]
hi @Vector1iu , we could do this in the future, will check with the team but feel free to contribute with a PR implementing it.

**Comment by handsomeZhuang** [comment]
Hi, Dear Development Team，
 We have recently used “faiss.index_factory(dim,'Flat', faiss.METRIC_Jaccard)" and index.search() to create index and query, then found the result is not precise. We also found that the implementation of faiss source is different from that of scipy lib,but scipy lib is the same with original Jaccard method. We look forward to your reply!
Best wishes~

**Comment by mlomeli1** [comment]
please open a new issue @handsomeZhuang

**Comment by handsomeZhuang** [comment]
issue has been created as 3521, please check it





------------------ Original ------------------
From: Maria Lomeli ***@***.***&gt;
Date: Wed,Jun 19,2024 10:55 PM
To: facebookresearch/faiss ***@***.***&gt;
Cc: handsomeZhuang ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [facebookresearch/faiss] support Jaccard metric in binary index(Issue #3134)





 
please open a new issue @handsomeZhuang

**Comment by mlomeli1** [comment]
yes, sorry you are right @handsomeZhuang , please bear with us, we will get back to you

**Comment by etiur** [comment]
Hello, Has this been implemented or corrected?
